### DISCUSSION TITLE
Upgrade from 3.8.16 to 3.13.6: runtime fails to load module files

### DISCUSSION THREAD

**Discussion by cbittencourt** [rabbitmq/rabbitmq-server#12102] [Other] [answered]
### Describe the bug

Getting the following error with the zip install of RabbitMq version 3.13.6 with OTP version 26.2.5.

Note: Adding `chcp 65001` provided the error and crash reports.

`> chcp 65001 && c:\rabbitmq\sbin\rabbitmq-plugins.bat enable rabbitmq_management --offline`
```
Active code page: 65001
=ERROR REPORT==== 22-Aug-2024::16:17:24.581000 ===
beam\beam_load.c(551): Error loading function elixir:start/2: op put_tuple u x:
  please re-compile this module with an Erlang/OTP 26 compiler or update your Erlang/OTP version


=CRASH REPORT==== 22-Aug-2024::16:17:24.581000 ===
  crasher:
    initial call: application_master:init/4
    pid: <0.86.0>
    registered_name: []
    exception exit: {bad_return,
                        {{elixir,start,[normal,[]]},
                         {'EXIT',
                             {undef,
                                 [{elixir,start,[normal,[]],[]},
                                  {application_master,start_it_old,4,
                                      [{file,"application_master.erl"},
                                       {line,293}]}]}}}}
      in function  application_master:init/4 (application_master.erl, line 142)
    ancestors: [<0.85.0>]
    message_queue_len: 1
    messages: [{'EXIT',<0.87.0>,normal}]
    links: [<0.85.0>,<0.44.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 987
    stack_size: 28
    reductions: 183
  neighbours:

ERROR! Failed to start Elixir.
error: {error,
           {elixir,
               {bad_return,
                   {{elixir,start,[normal,[]]},
                    {'EXIT',
                        {undef,
                            [{elixir,start,[normal,[]],[]},
                             {application_master,start_it_old,4,
                                 [{file,"application_master.erl"},
                                  {line,293}]}]}}}}}}
```

The same installation process works with OTP 23.3 and RabbitMq 3.8.16.

If I'm doing something wrong, please advise and I apologize for the invalid report.

### Reproduction steps

1. Install https://github.com/erlang/otp/releases/download/OTP-26.2.5/otp_win32_26.2.5.exe
2. Set ERLANG_HOME environment variable `[Environment]::SetEnvironmentVariable("ERLANG_HOME", "c:\erlang", "Machine")`
3. Download RabbitMq ZIP and extract to c:\rabbitmq `https://github.com/rabbitmq/rabbitmq-server/releases/download/v3.13.6/rabbitmq-server-windows-3.13.6.zip`
4. Set RabbitMq Configuration
```
[Environment]::SetEnvironmentVariable("RABBITMQ_HOME", "c:\rabbitmq", "Machine")
[Environment]::SetEnvironmentVariable("RABBITMQ_CONFIG_FILE", "c:\rabbitmq\etc\rabbitmq.conf", "Machine")
"loopback_users = none" | Out-File "c:\rabbitmq\etc\rabbitmq.conf"
```
5. Try any rabbitmq command (`chcp 65001 && c:\rabbitmq\sbin\rabbitmq-plugins.bat enable rabbitmq_management --offline`

### Expected behavior

Install works successfull.

**Comment by michaelklishin** [comment, marked as answer]
RabbitMQ 3.8.16 [has never supported Erlang 26](https://www.rabbitmq.com/docs/which-erlang).

RabbitMQ 3.13 is tested on Erlang 26 on Windows multiple times a day as part of our CI. There
are no similar reports from Windows users, so something is specific to your system and it's up to you to investigate what.

@cbittencourt your setup ends up with RabbitMQ 3.8 trying to start on Erlang 26 or [vice versa](https://www.rabbitmq.com/docs/windows-configuration#multiple-erlang-versions), if I were to guess, and since in the 3.8 era Elixir (if not RabbitMQ itself) was compiled on Erlang 21 or earlier, Erlang 25's runtime cannot load its module files. At least that's my best guess based on the message.

**Reply by michaelklishin** [reply]
One good example of how modern RabbitMQ and Erlang are provisioned is the [Chocolatey package](https://github.com/rabbitmq/chocolatey-package/tree/3.13.x/tools) which is open source and can be easily inspected.

In fact, you are welcome to simply use that instead of rolling your own automation.
